xdsl/dialects/riscv.py

```diff
@@ -0,0 +1,765 @@
+from __future__ import annotations
+from dataclasses import dataclass, field
+
+from typing import Union, TypeVar, Type
+
+from xdsl.ir import Dialect, MLContext, Operation, Data
+from xdsl.irdl import irdl_op_definition, irdl_attr_definition, builder, AnyOf
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.dialects.builtin import StringAttr, IntegerAttr, OpAttr
+
+
+@dataclass(frozen=True)
+class Register:
+    """A riscv register."""
+
+    index: int = field()
+    """The register index. Can be between 0 and 31."""
+
+    abi_names = {
+        "zero": 0,
```

Comment:
Feels a tiny bit off to me to have a list of integers typed out like this as opposed to a `list` or `ordereddict`, but I guess it's not a big deal

Reply:
I think I like this more than `zip(...)` or something else that has implicit values because it's clearer what value maps to what.

Reply:
Yeah that's a good point